### PR TITLE
Updates in lesson2.md

### PR DESCRIPTION
**Pull request by pratyakshajha**
Added notes made by #math-help SLs for specific topics - gradient descent, cross entropy and sigmoid's derivative.

### CODE REVIEW

**Comment by ishgirwan** [comment]
This is nice. Thanks 👍
